### PR TITLE
Support Aqua S2 Lock

### PR DESCRIPTION
**Pull request by dzungpv**
After long time, i finally back to this

### CODE REVIEW

**Comment by dzungpv** [comment]
Could you add this to master? https://github.com/Koenkk/zcl-packet/pull/6, without it this pull will not work

**Comment by dzungpv** [comment]
@Koenkk with new code i make very little change to zcl-packet lib, please review and merge both. Thanks

File: devices.js [hunk 1]
```diff
@@ -461,7 +461,16 @@ const devices = [
...
-
+    {
+        zigbeeModel: ['lumi.lock.acn02'],
+        model: 'ZNMS12LM',
+        description: 'Aqara S2 Lock',
+        supports: 'report: open, close, operation',
+        vendor: 'Xiaomi',
+        fromZigbee: [fz.ZNMS12LM_closuresDoorLock_change, fz.ZNMS12LM_genBasic_change,
+            fz.ignore_doorlock_change, fz.ignore_basic_change],
+        toZigbee: [],
```
Comment (Koenkk) [inline devices.js]: Can the lock also be controlled?

File: converters/fromZigbee.js [hunk 2]
```diff
@@ -2682,7 +2682,115 @@ const converters = {
...
+                    result.state = 'LOCK';
+                    result.reverse = 'LOCK';
+                } else if (command === '0111') {
+                    result.event = lockStatusLookup[5];
+                    result.state = 'LOCK';
+                    result.reverse = 'UNLOCK';
+                } else if (command === '0b00') {
+                    result.event = lockStatusLookup[12];
+                    result.state = 'UNLOCK';
+                    result.reverse = 'UNLOCK';
```
Comment (Koenkk) [inline converters/fromZigbee.js]: What does `reverse` mean?

File: converters/fromZigbee.js [hunk 3]
```diff
@@ -2682,7 +2682,115 @@ const converters = {
...
+                if (command === '0301') {
+                    result.event = lockStatusLookup[4];
+                    result.state = 'UNLOCK';
+                    result.reverse = 'UNLOCK';
+                } else if (command === '0311') {
+                    result.event = lockStatusLookup[4];
+                    result.state = 'LOCK';
+                    result.reverse = 'UNLOCK';
+                } else if (command === '0205') {
+                    result.event = lockStatusLookup[3];
```
Comment (Koenkk) [inline converters/fromZigbee.js]: Could you rename `event` to `action`? In this way we can use the home assistant action sensor which is already in place.

File: devices.js (hunk 1 above)
Comment (dzungpv) [reply devices.js]: No, it is work like a sensor

File: converters/fromZigbee.js (hunk 2 above)
Comment (dzungpv) [reply converters/fromZigbee.js]: "reverse"  mean lock inside, and outside open disable all, i use original word from xiaomi app

File: converters/fromZigbee.js (hunk 3 above)
Comment (dzungpv) [reply converters/fromZigbee.js]: Yes, i have already changed it

File: converters/fromZigbee.js (hunk 2 above)
Comment (Koenkk) [reply converters/fromZigbee.js]: I don't understand, is it like https://www.youtube.com/watch?v=Sz-2VDfClnM?

Comment (dzungpv) [reply converters/fromZigbee.js]: @Koenkk In the picture below, Yellow is like what you see in the video, Blue is the "Locked "state, Red is "Reverse locked" state, Aqara S2 is much complex. "Reverse locked" can only activated manual  inside, when you activate it, open from outside by key, finger or password all disable, you can see "Reverse locked" ativate in this https://youtu.be/bPvoQN51_5I?t=421 at 7:01
Screenshot_20190628-201148_Mi Home

**Comment by Koenkk** [comment]
Thanks!

**Pull request merged**